edx/frontend-app-library-authoring
# [BD-14] [SE-2939] Complex library authoring

This pull request enables Complex Library authoring. It replaces the existing library detail page. It adds buttons at the bottom for varied component addition and also features search and block type filtering capabilities.

This PR also contains several improvements to testing infrastructure developed while tracking down bugs caused by previous technical debt.

For convenience, a review copy of this PR is available here: https://github.com/open-craft/frontend-app-library-authoring/pull/5

**Discussions**: https://docs.google.com/document/d/1hRrSmWWlGHhi-bh9AIlpSuLtJFnn2GpavNe8OQxthbE/edit

**Dependencies**: https://github.com/edx/edx-platform/pull/25368

**Screenshots**:

![image](https://user-images.githubusercontent.com/1099483/96354526-49437680-109d-11eb-9c93-5f8315c5cc80.png)

![image](https://user-images.githubusercontent.com/1099483/96354537-60826400-109d-11eb-9cbc-91259e02c84f.png)

![image](https://user-images.githubusercontent.com/1099483/96354554-827be680-109d-11eb-9bda-f1df6627eb0e.png)

![image](https://user-images.githubusercontent.com/1099483/96354563-99223d80-109d-11eb-9cda-ae115fe26e63.png)

**Merge deadline**: No specific deadline, but sooner is better than later.

**Testing instructions**:

1. Set up https://github.com/edx/edx-platform/pull/25368
2. Create and visit a library of type Complex.
3. Add each type of block by its convenient component button.
4. Click the edit button on a block to verify it takes you to the edit page.
5. Try out all the editing features for blocks which support editing (such as drag and drop.) Check over OLX and asset uploads to  verify they still work.
6. Edit the library's details and verify you cannot change its type if there are unpublished changes or deletes.
7. Verify you can still properly manage user access.
8. Search the library for terms which appear in the blocks' titles, for instance.
9. Filter the blocks by type. Try a specific type, 'All Types' and 'Other Types' to make sure they all behave sanely.

**Reviewers**
